### PR TITLE
Serializeable API Schema

### PR DESCRIPTION
**Pull request by jaredoconnell**
This adds a basic schema.

Supports:
- keepalive
- basic auth
- plugin and protocol data

There is an instruction type that should allow us to not deserialize it all at once, and instead deserialize that part separately once the type is known.

### CODE REVIEW

File: src/api/schema/instructions.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    pub instruction_type: CoreInstructionType,
+    pub payload: Box<RawValue>, // or &'a RawValue
+}
+
+
+#[cfg(test)]
+mod tests {
+    
+
+}
```
Comment (kdvalin) [inline src/api/schema/instructions.rs]: If we're not testing anything in this file, remove the `tests` module

File: src/api/schema/auth.rs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        let original = AuthAccountInstruction {
+            used_authmethod: AuthMethod {
+                name: "test".to_string(),
+                fields: vec![],
+            },
+
+        };
+        let serialized = serde_json::to_string(&original).unwrap();
+
+        println!("serialized AuthAccountInstruction = {}", serialized);
```
Comment (kdvalin) [inline src/api/schema/auth.rs]: Replace println! with Rust's `log` crate (using debug! or trace! more likely)

**Review by kdvalin — CHANGES_REQUESTED** [review]

File: src/api/schema/auth.rs (hunk 2 above)
Comment (jaredoconnell) [reply src/api/schema/auth.rs]: Looks like it's not supported in test cases. https://stackoverflow.com/questions/67087597/is-it-possible-to-use-rusts-log-info-for-tests

Comment (jaredoconnell) [reply src/api/schema/auth.rs]: Fixed as discussed in external discussion.

**Review by kdvalin — APPROVED** [review]

**Pull request merged**